% \VignetteIndexEntry{ReactomePA - an R package for Reactome Pathway Analysis}
% \VignettePackage{ReactomePA}
% \VignetteEngine{knitr::knitr}

% To compile this document, run the commands within R
% require(knitr); knit2pdf("ReactomePA.Rnw")

\documentclass[12pt]{article}


<<knitr, echo=FALSE, results="hide">>=
knitr::opts_chunk$set(tidy         = FALSE,
               out.truncate = 80,
               out.lines    = 6,
               dev          = 'pdf',
               include      = TRUE,
               fig.width    = 6,
               fig.height   = 6,
               resolution   = 100,
               message      = FALSE)
@ 

<<style-Sweave, eval=TRUE, echo=FALSE, results="asis">>=
BiocStyle::latex()
@ 


<<loadPKG, echo=FALSE>>=
library(DOSE)
library(ReactomePA)
library(clusterProfiler)
@




\author{Guangchuang Yu \\[1em]
  \small{School of Public Health} \\
  \small{The University of Hong Kong} \\
  \small{\email{guangchuangyu@gmail.com}}
}


\title{Reactome Pathway Analysis}

\begin{document}
\maketitle

<<options,results='hide',echo=FALSE>>=
options(digits=3, width=80, prompt=" ", continue=" ")
@

\newpage
\tableofcontents
\newpage

\section{Introduction}

This package is designed for reactome pathway-based analysis. Reactome
is an open-source, open access, manually curated and peer-reviewed
pathway database.

\section{Pathway Enrichment Analysis}

Enrichment analysis is a widely used approach to identify biological
themes. Here, we implement hypergeometric model to assess whether the
number of selected genes associated with reactome pathway is larger
than expected. The p values were calculated based the hypergeometric model \cite{boyle2004},

<<load sample data>>=
require(DOSE)
data(geneList)
de <- names(geneList)[abs(geneList) > 1.5]
head(de)
require(ReactomePA)
x <- enrichPathway(gene=de,pvalueCutoff=0.05, readable=T)
head(summary(x))
@

\subsection{Visualize enrichment result}
We implement bar plot, enrichment map and category-gene-network for visualization. It is very common to visualize the enrichment result in bar or pie chart. We believe the pie chart is misleading and only provide bar chart.
<<barplot, fig.cap="barplot of Reactome Pathway enrichment result.", fig.align="center", fig.height=5, fig.width=8, out.width="0.8\\textwidth", fig.pos="h">>=
barplot(x, showCategory=8)
@

Enrichment map can be viusalized by \Rfunction{enrichMap} :
<<enrichMap, fig.cap="enrichment map of enrichment result", fig.align="center", fig.height=16, fig.width=20, out.width="0.9\\textwidth", fig.pos="h">>=  
enrichMap(x)
@ 

In order to consider the potentially biological complexities in which a gene may belong to multiple annotation categories, we developed \Rfunction{cnetplot} function to extract the complex association between genes and diseases.
<<cnetplot, fig.cap="cnetplot of Reactome Pathway enrichment result.", fig.align="center", fig.height=18, fig.width=20, out.width="0.9\\textwidth", fig.pos="h">>=
cnetplot(x, categorySize="pvalue", foldChange=geneList)
@

\subsection{Comparing enriched reactome pathways among gene clusters with clusterProfiler}
We have developed an \textsf{R} package \Biocpkg{clusterProfiler} \cite{yu_clusterprofiler:_2012} for comparing biological themes among gene clusters. \Biocpkg{ReactomePA} works fine with \Biocpkg{clusterProfiler} and can compare biological themes at reactome pathway perspective.
 
<<clusterProfiler, fig.cap="ReactomePA with clusterProfiler.", fig.align="center", fig.height=8, fig.width=11, out.width="0.9\\textwidth", fig.pos="h">>= 
require(clusterProfiler)
data(gcSample)
res <- compareCluster(gcSample, fun="enrichPathway")
plot(res)
@

\section{Gene Set Enrichment Analysis}
A common approach in analyzing gene expression profiles was identifying differential expressed genes that are deemed interesting. The \Rfunction{enrichPathway} function we demonstrated previously were based on these differential expressed genes. This approach will find genes where the difference is large, but it will not detect a situation where the difference is small, but evidenced in coordinated way in a set of related genes. Gene Set Enrichment Analysis (GSEA) directly addressed this limitation. All genes can be used in GSEA; GSEA aggregates the per gene statistics across genes within a gene set, therefore making it possible to detect situations where all genes in a predefined set change in a small but coordinated way. 

<<GSEA analysis>>=
y <- gsePathway(geneList, nPerm=100, 
                minGSSize=120, pvalueCutoff=0.05, 
                pAdjustMethod="BH", verbose=FALSE)
res <- summary(y)
head(res)
@

\subsection{Visualize GSEA result}
<<enrichMap2, fig.cap="enrichment map of gsea result", fig.align="center", fig.height=16, fig.width=16, out.width="0.9\\textwidth", fig.pos="h">>=  
enrichMap(y)
@

<<gseaplot, fig.cap="plotting gsea result", fig.align="center", fig.height=6, fig.width=8, out.width="0.6\\textwidth", fig.pos="h">>=  
gseaplot(y, geneSetID = "1280215")
@

\section{Pathway Visualization}

In \Biocpkg{ReactomePA}, we also implemented \Rfunction{viewPathway} to visualized the pathway.
<<viewPathway, fig.cap="Reactome Pathway visualization.", fig.align="center", fig.height=16, fig.width=16, out.width="0.9\\textwidth", fig.pos="h">>=
viewPathway("E2F mediated regulation of DNA replication", readable=TRUE, foldChange=geneList)
@


\section{Session Information}

Here is the output of \Rcode{sessionInfo()} on the system on which this document was compiled:

<<sessInfo, results='asis', echo=FALSE>>=
toLatex(sessionInfo())
@


\bibliography{ReactomePA}
\end{document}

